src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -171,10 +195,21 @@ private PartitionedRateLimiter<HttpContext> CreateEndpointLimiter()
         // If we have a policy for this endpoint, use its partitioner. Else use a NoLimiter.
         return PartitionedRateLimiter.Create<HttpContext, DefaultKeyType>(context =>
         {
-            var name = context.GetEndpoint()?.Metadata.GetMetadata<IRateLimiterMetadata>()?.PolicyName;
+            DefaultRateLimiterPolicy? policy;
+            var enableRateLimitingAttribute = context.GetEndpoint()?.Metadata.GetMetadata<EnableRateLimitingAttribute>();
+            if (enableRateLimitingAttribute is null)
+            {
+                return RateLimitPartition.GetNoLimiter<DefaultKeyType>(_defaultPolicyKey);
+            }
+            policy = enableRateLimitingAttribute.Policy;
+            if (policy is not null)
+            {
+                return policy.GetPartition(context);
+            }
+            var name = enableRateLimitingAttribute.PolicyName;
             if (name is not null)
```

Comment:
Should the else case throw here? It'd be an odd situation to be able to get this far and have a null policy name, and we probably shouldn't default to no limit.

Reply:
Technically it should be impossible for both the policyName & policy to be null on a `EnableRateLimitingAttribute`, but it couldn't hurt to defensively throw for that case.